test/utils/test_utils_backend.dart

```diff
@@ -0,0 +1,185 @@
+import 'dart:async';
+import 'dart:io';
+import 'package:http/http.dart' as http;
+import 'package:http/retry.dart';
+import 'dart:convert';
+
+import 'package:icure_medical_device_dart_sdk/mappers/code_coding.dart';
+
+abstract class TestBackend {
+
+  abstract final String iCureUser;
+  abstract final String iCurePwd;
+  abstract final String iCureURL;
+
+  Future<void> init();
+  Future<void> shutdown();
+
+}
+
+class RemoteTestBackend implements TestBackend {
+
+  static TestBackend? _instance;
+  @override final String iCurePwd;
+  @override final String iCureURL;
+  @override final String iCureUser;
+
+  RemoteTestBackend(this.iCureUser, this.iCurePwd, this.iCureURL);
+
+  static TestBackend getInstance(String iCureUser, String iCurePwd, [String iCureURL = "https://kraken.icure.dev"]) {
+    if (RemoteTestBackend._instance == null) {
+      RemoteTestBackend._instance = RemoteTestBackend(iCureUser, iCurePwd, iCureURL);
+    }
+    return RemoteTestBackend._instance!;
+  }
+
+  @override
+  Future<void> init() async { }
+
+  Future<void> shutdown() async { }
+
+}
+
+class DockerTestBackend implements TestBackend {
+
+  static TestBackend? _instance;
+  final int DB_PORT;
+  final int AS_PORT;
+  final String couchdbUser;
+  final String couchdbPassword;
+  @override final String iCurePwd;
+  @override final String iCureURL;
+  @override final String iCureUser;
+
+  DockerTestBackend(this.DB_PORT, this.AS_PORT, this.couchdbUser, this.couchdbPassword, this.iCureUser, this.iCurePwd, this.iCureURL);
```

Comment:
Is the DockerTestBackend working properly for Dart ? 
If yes, then it can stay in your PR, if not you can remove it from now and we'll add it later when we need it

Reply:
It does work, but I ended up not using it so I think I can remove it